### ISSUE TITLE
Can we export all types from the package?

### ISSUE THREAD

**Issue by 4R53N** [makenotion/notion-sdk-js#280]
Title says it all. There are types like `RichTextItemResponse` and `BlockObjectResponse` that would be useful to have so we could create utils in our integrations that perform type checking

**Comment by seancdavis** [comment]
This would also help me a lot. I've currently copied the `api-endpoints.ts` file to my project and exported all types, and it's made type checking a much more pleasant process.

Another related suggestion is to use types for all objects, rather than embedding objects. For example, instead of this:

```ts
export type BlockObjectResponse =
  | {
      type: "paragraph";
      // ...
    }
  | {
      type: "heading_1";
      // ...
    }
  // ...
```

Do this:

```ts
export type ParagraphBlockResponse = {
  type: "paragraph";
  // ...
}

export type Heading1BlockResponse = {
  type: "heading_1";
  // ...
}

export type BlockObjectResponse = {
  ParagraphBlockResponse | Heading1BlockResponse | // ...
}
```

This way, if I've done the checking and am confident I'm working with a paragraph, then I can take advantage of the types this library is generating. Otherwise, I'm getting almost no benefit from the typings.

**Comment by transitive-bullshit** [comment]
I also would love this as I'm working on adding a compatibility layer between the official and unofficial Notion API types for [react-notion-x](https://github.com/NotionX/react-notion-x).

Exposing the types that an API uses is commonplace for npm packages so that consumers can make use of these types when building on top of it.

**Comment by hahnbeelee** [comment]
Here's another ugly work-around.

For BlockResponses:
```
type AdjustedBlockResponse <T> = Partial<T>
   & { type?: string }

const response: AdjustedBlockResponse<GetBlockResponse> = await notion.blocks.retrieve({
            block_id: blockId,
        });
```

For PageResponses:
```
type AdjustedPageResponse <T> = Partial<T>
   & { properties?: {
       title?: { title: {
           text: {
               content: string
           }
       }[]}
   }}

const response: AdjustedPageResponse<GetPageResponse> = await notion.pages.retrieve({ page_id: pageId });
```

**Comment by robinscholz** [comment]
Would also need this!

**Comment by e-e-e** [comment]
This would be very useful indeed. In trying to work around this issue https://github.com/makenotion/notion-sdk-js/issues/288 I have found myself having to extract the typing information from the response types because they are not exported separately. Its quite difficult.

In case its useful to others - this is the hack that I have been using in order to get the individual types :
```ts
import { QueryDatabaseResponse } from '@notionhq/client/build/src/api-endpoints';

type AllKeys<T> = T extends never ? never : keyof T;
type OptionalKeys<T> = Exclude<AllKeys<T>, keyof T>;
type Idx<T, K extends PropertyKey, D = never> = T extends never
  ? never
  : K extends keyof T
  ? T[K]
  : D;

type Widen<T> = {
  [K in OptionalKeys<T>]?: Idx<T, K>;
} & { [K in keyof T]: T[K] };

// I found widening the Result type is needed because the type is defined 
// as a union which effectively omits many of the properties from the type
type NotionDatabaseQueryResult = Widen<
  QueryDatabaseResponse['results'][number]
>;
type NotionPageProperties = NonNullable<
  NotionDatabaseQueryResult['properties']
>;
export type NotionPagePropertiesValues = NotionPageProperties[string];

export type InferFromType<
  T extends { type: string },
  P extends PropertyKey
> = T extends {
  type: P;
}
  ? T
  : never;

export type NotionPropertyType = NotionPagePropertiesValues['type'];

// This extracts only the type of the 'people' object from the `properties` type.
export type NotionPeopleProperty = InferFromType<
  NotionPagePropertiesValues,
  'people'
>;

export type WidenedNotionPeopleResponseObject = Widen<
  NotionPeopleProperty['people'][number]
>;

```

This is very convoluted, expensive for the tsc, and completely avoidable if the types were isolated exported individually.

**Comment by itsdarrylnorris** [comment]
Same here, I am writing service layer on top of this library, and it will be nice to have the types. Currently I am copying them over, which is not great.

**Comment by LumaKernel** [comment]
I also copied all and split types by hand. We, not members, may not make PRs because it seems auto-generated. Split types are very useful to create my own rendering. BTH, this lib is awesome! Thanks.

**Comment by JamesNZL** [comment]
Yes please!

Currently I'm having to use workarounds such as using the `Extract` utility for `TimeZoneRequest`, `EmojiRequest` and `RichTextItemResponse`:

```ts
export type valueof<T> = T[keyof T];
export type ArrayElement<ArrayType extends readonly unknown[]> = ArrayType extends readonly (infer ElementType)[] ? ElementType : never;

type DateRequest = NonNullable<NonNullable<Extract<valueof<CreatePageParameters['properties']>, { type?: 'date'; }>['date']>>;
export type TimeZoneRequest = DateRequest['time_zone'];

export type EmojiRequest = Extract<CreatePageParameters['icon'], { type?: 'emoji'; }>['emoji'];

export type RichTextItemResponse = ArrayElement<Extract<Extract<ArrayElement<SearchResponse['results']>, { object: 'database'; }>, { title: unknown; }>['title']>;
```

**Comment by wcauchois** [comment]
Hi, I recently made some changes to name many more response types and export them from this package in #319, so I am going to close this issue. These changes are available in [version 2.1.0](https://github.com/makenotion/notion-sdk-js/releases/tag/v2.1.0) of the SDK.

Note that I have not exported _all_ types per the request from this issue, but every exported type requires a stability guarantee from us so I've been somewhat judicious. You can still extract types using TypeScript helpers like type indexes and Extract<>, where necessary. If there are any more specific issues around types please open a new GitHub issue, thank you!
